src/Donors/Endpoints/ListDonors.php

```diff
@@ -65,35 +78,153 @@ public function registerRoute()
+                        'type' => 'string',
+                        'required' => false,
+                        'sanitize_callback' => 'sanitize_text_field',
+                    ],
+                    'sortDirection' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'enum' => [
+                            'asc',
+                            'desc'
+                        ],
+                    ],
+                    'locale' => [
+                        'type' => 'string',
+                        'required' => false,
+                        'default' => get_locale(),
+                    ],
                 ],
             ]
         );
     }
 
     /**
-     * @param WP_REST_Request $request
+     * @unreleased Change this to use the new ListTable class
      * @since 2.20.0
      *
+     * @param WP_REST_Request $request
+     *
      * @return WP_REST_Response
      */
     public function handleRequest(WP_REST_Request $request): WP_REST_Response
     {
-        $data = [];
-        $controller = new DonorsRequestController($request);
-        $donors = $controller->getDonors();
-        $donorsCount = $controller->getTotalDonorsCount();
+        $this->request = $request;
+        $this->listTable = give(DonorsListTable::class);
+
+        $donors = $this->getDonors();
+        $donorsCount = $this->getTotalDonorsCount();
         $pageCount = (int)ceil($donorsCount / $request->get_param('perPage'));
 
-        foreach ($donors as $donor) {
-            $data[] = DonorResponseData::fromObject($donor)->toArray();
-        }
+        $this->listTable->items($donors, $this->request->get_param('locale'));
 
         return new WP_REST_Response(
             [
-                'items' => $data,
+                'items' => $this->listTable->getItems(),
                 'totalItems' => $donorsCount,
                 'totalPages' => $pageCount
             ]
         );
     }
+
+    /**
+     * @unreleased Replace Query Builder with Donations model
+     * @since 2.21.0
+     *
+     * @return array
+     */
+    public function getDonors(): array
+    {
+        $page = $this->request->get_param('page');
+        $perPage = $this->request->get_param('perPage');
+        $sortColumns = $this->listTable->getSortColumnById($this->request->get_param('sortColumn') ?: 'id');
+        $sortDirection = $this->request->get_param('sortDirection') ?: 'desc';
+
+        $query = give()->donors->prepareQuery();
+        $query = $this->getWhereConditions($query);
+
+        foreach ($sortColumns as $sortColumn) {
+            $query->orderBy($sortColumn, $sortDirection);
+        }
+
+        $query->limit($perPage)
+            ->offset(($page - 1) * $perPage);
+
+        $donors = $query->getAll();
+
+        if (!$donors) {
+            return [];
+        }
+
+        return $donors;
+    }
+
+    /**
+     * @unreleased Replace Query Builder with Donations model
```

Comment:
looks like a typo: `Donations` should be `Donor`

Reply:
Resolved: 6436ae9e04ee7ca53e393f8cd4bd8438256d2ff3